src/AppBundle/Entity/CasRec.php

```diff
@@ -227,20 +227,37 @@ public function getCorref()
      *
      * @param string $typeOfRep e.g. opg103
      * @param string $corref    e.g. l3, l3g
+     * @param string $userRoleName    e.g. ROLE_PA
      *
      * @return string Report::TYPE_*
      */
-    public static function getTypeBasedOnTypeofRepAndCorref($typeOfRep, $corref)
+    public static function getTypeBasedOnTypeofRepAndCorref($typeOfRep, $corref, $userRoleName)
     {
         $typeOfRep = trim(strtolower($typeOfRep));
         $corref = trim(strtolower($corref));
 
-        if (Report::ENABLE_103 && in_array($corref, ['l3', 'l3g', 'a3']) && $typeOfRep === 'opg103') {
-            return Report::TYPE_103;
-        } elseif (Report::ENABLE_104 && $corref === 'hw' && $typeOfRep === '') {
-            return Report::TYPE_104;
+        if ($userRoleName === User::ROLE_LAY_DEPUTY) {
+            if (in_array($corref, ['l3', 'l3g', 'a3']) && $typeOfRep === 'opg103') {
+                return Report::TYPE_103;
+            }
+            if (Report::ENABLE_104 && $corref === 'hw' && $typeOfRep === '') {
+                return Report::TYPE_104;
+            }
+
+            return Report::TYPE_102; //default for Lay
+        }
+
+        if (in_array($userRoleName, [User::ROLE_PA, User::ROLE_PA_TEAM_MEMBER, User::ROLE_PA_ADMIN])) {
+            if (in_array($corref, ['l3', 'l3g', 'a3']) && $typeOfRep === 'opg103') {
+                return Report::TYPE_103_6;
+            }
+            if (Report::ENABLE_104 && $corref === 'hw' && $typeOfRep === '') {
+                return Report::TYPE_104_6;
+            }
+
+            return Report::TYPE_102_6; //default for PA
```

Comment:
Id be tempted to throw an exception if the type cannot be determined by logic. Rather than have incorrect report types in the database where we wouldnt neccessarily be aware

Reply:
there still is a default to 102 for each user role, but we can think of having an exception once the CSV fields are more clear